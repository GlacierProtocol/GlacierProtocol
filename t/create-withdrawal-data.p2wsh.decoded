{
  "txid": "3b6c2194e62d1f72b200d30745a5a12deb8196e4f1247d3b01e3f14496abb7e0",
  "hash": "05641b3ea4389210d6607091cd6ba2cf909630228d211e3ac301ad0f32fc665b",
  "version": 2,
  "size": 370,
  "vsize": 154,
  "weight": 616,
  "locktime": 0,
  "vin": [
    {
      "txid": "8fb83e5a38315112c25bb1cf32073f14ec1519bdb6fe6ce8e1b2e054e55d4c7c",
      "vout": 0,
      "scriptSig": {
        "asm": "",
        "hex": ""
      },
      "txinwitness": [
        "",
        "304402204d7646f74a931211b32fa7d999d8533844f8582702085819e0d93026e8bf65c902206aa350f1cff87e2960e7dd06eb04dbc97be637452da37aa7975131c6dc8970fb01",
        "3044022047d2c218398278e5d98869fcc6a803fb909f5928e794e894c0c4ebaab84e5d9702204a83e79812cc5d502db0b0d8e0709d3d2aacfb1e998920d93cb0d7a5bcd3550701",
        "522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae"
      ],
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.04999606,
      "n": 0,
      "scriptPubKey": {
        "asm": "0 a410c764ac8da12b578a06027f11e02f07b7e1d3",
        "hex": "0014a410c764ac8da12b578a06027f11e02f07b7e1d3",
        "reqSigs": 1,
        "type": "witness_v0_keyhash",
        "addresses": [
          "tb1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnplwvac"
        ]
      }
    }
  ]
}
