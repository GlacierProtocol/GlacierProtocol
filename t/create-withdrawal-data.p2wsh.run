#!/bin/bash
set -e

../../glacierscript.py --testnet=$1 create-withdrawal-data << INPUT
y
y
y
y
y
y
tb1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q8hz22g
522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae
tb1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnplwvac
1
020000000001016fac4e4851a864b8f2621bb031a61e112ca75898fd8f11a3247b7bf0fce07bfc0000000017160014b787e8916e12d6ab16f3b52d63f36950be81faeffdffffff02404b4c000000000022002065c863c750331cc03efb4daca217e7925cff87f5e9ca9927564234be0d6278e8dba19e00000000001600142eaef37b593178cced910e6a2e36f1c2db328c8402473044022022e32a7d67ad3dc1c247210f58a261c9d071d3c4e111b47e9a32df40f952dceb02200c2a782566fff6f67457689621f920e8a54b6ce32e2af93211464f0cc866c3f301210379b5b02900f37f9507110d1b5b20e1e01304ee8772781bb5e3f7dae9d258bbb900000000
2
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4
2
y

y
INPUT
