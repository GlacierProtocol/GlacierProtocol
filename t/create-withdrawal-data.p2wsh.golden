Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.05000000 BTC

How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00000394 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00000394, you have 0.04999606 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to tb1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnplwvac (leave blank to withdraw all funds stored in these unspent transactions): 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.05000000 BTC in unspent supplied transactions
0 BTC going back to cold storage address tb1qvhyx836sxvwvq0hmfkk2y9l8jfw0lpl4a89fjf6kgg6turtz0r5q8hz22g
0.04999606 BTC going to destination address tb1q5sgvwe9v3ksjk4u2qcp87y0q9urm0cwnplwvac
Fee amount: 0.00000394

Signing with private keys: 
cQCrT9Ncs9729ao7jbmAWrD9z7tF64s2yKzmD6nkiLAi9sXVZWAn
cP65UeSDZPiTLB6CBwasWv9oJYEjRgQXhswfwcT9HscEKDcEbgy4


Confirm? (y/n): 
Calculating transaction...


Sufficient private keys to execute transaction?
True

Raw signed transaction (hex):
020000000001017c4c5de554e0b2e1e86cfeb6bd1915ec143f0732cfb15bc2125131385a3eb88f0000000000ffffffff01b6494c0000000000160014a410c764ac8da12b578a06027f11e02f07b7e1d3040047304402204d7646f74a931211b32fa7d999d8533844f8582702085819e0d93026e8bf65c902206aa350f1cff87e2960e7dd06eb04dbc97be637452da37aa7975131c6dc8970fb01473044022047d2c218398278e5d98869fcc6a803fb909f5928e794e894c0c4ebaab84e5d9702204a83e79812cc5d502db0b0d8e0709d3d2aacfb1e998920d93cb0d7a5bcd35507018b522103d14ddcfb6817f5579695bbb3eb3e185887bf2942b031e6f716343b8fe7e9e8e221028fcd46f8614b2cbf318096968242a1e22bcfb6d8f2b6dc939c8c27c347b2937b210315acb550120f4cdcb460d5c49080ab508ca4ccd8dedecac22feeac8cd017d4c121022b063ee2f22f9e1982c140fe778672c683111a796dcccdbd47bb65e3d608a98354ae00000000

Transaction fingerprint (md5):
3960f5cc65b55d6452b026b49af75e8c
QR code for transaction written to transaction.png
